packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java

```diff
@@ -79,13 +84,91 @@ public void createRule()
+        restClient.assertLastError().containsSummary("fake-id was not found");
+    }
+
+    /** Check creating a rule in a non-existent rule set returns an error. */
+    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
+    public void createRuleInNonExistentRuleSet()
+    {
+        STEP("Try to create a rule in non-existent rule set.");
+        RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("ruleName");
+
+        restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingRuleSet("fake-id").createSingleRule(ruleModel);
+
+        restClient.assertStatusCodeIs(NOT_FOUND);
+        restClient.assertLastError().containsSummary("fake-id was not found");
+    }
+
+    /** Check we can create two rules with the same name. */
+    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
+    public void duplicateRuleNameIsAcceptable()
+    {
+        RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("duplicateRuleName");
+
+        STEP("Create two identical rules");
+        RestRuleModel ruleA = restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().createSingleRule(ruleModel);
+        RestRuleModel ruleB = restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet().createSingleRule(ruleModel);
+
+        // Check that the names are the same but the ids are different.
+        ruleA.assertThat().field("name").is(ruleB.getName());
+        ruleA.assertThat().field("id").isNot(ruleB.getId());
+    }
+
+    /** Check that a user without permission to view the folder cannot create a rule in it. */
+    public void requirePermissionToCreateRule()
+    {
+        STEP("Create a user and use them to create a private site containing a folder");
+        UserModel privateUser = dataUser.createRandomTestUser();
+        SiteModel privateSite = dataSite.usingUser(privateUser).createPrivateRandomSite();
+        FolderModel privateFolder = dataContent.usingUser(privateUser).usingSite(privateSite).createFolder();
+
+        STEP("Try to use a different user to create a rule in the private folder");
+        RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("ruleName");
+
+        restClient.authenticateUser(user).withCoreAPI().usingNode(privateFolder).usingDefaultRuleSet().createSingleRule(ruleModel);
+
+        restClient.assertStatusCodeIs(FORBIDDEN);
+        restClient.assertLastError().containsSummary("Cannot read from this node");
+    }
+
+    /** Check we can't create a rule under a document node. */
+    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
+    public void tryToCreateRuleUnderDocument()
+    {
+        STEP("Create a document.");
+        FileModel fileModel = dataContent.usingUser(user).usingSite(site).createContent(getRandomFileModel(TEXT_PLAIN));
+
+        RestRuleModel ruleModel = new RestRuleModel();
+        ruleModel.setName("");
```

Comment:
I assume missing name will be covered later, ie. as per PR notes, more changes to come in follow-on PRs:

> Omit the rule name (requires a production code change)

Reply:
This looks like a copy/paste mistake. This test shouldn't have an empty rule name.